RockPaperScissors/RockPaperScissors/main.swift

```diff
@@ -1,10 +1,86 @@
 //
 //  RockPaperScissors - main.swift
-//  Created by yagom. 
+//  Created by Lingo, 우롱차.
 //  Copyright © yagom academy. All rights reserved.
 // 
 
 import Foundation
 
-print("Hello, World!")
+enum GameDisplaySetting: String {
+    case menu = "가위(1), 바위(2), 보(3)! <종료 : 0> :"
+    case gameEnd = "게임 종료"
+    case gameWin = "이겼습니다!"
+    case gameLose = "졌습니다!"
+    case gameDraw = "비겼습니다!"
+    case gameWrongInput = "잘못된 입력입니다. 다시 시도해주세요."
+}
 
+enum GameOptions: Int {
+    case gameEnd = 0
+    case scissor = 1
+    case lock = 2
+    case paper = 3
+}
+
+enum GameResult {
+    case win
+    case lose
+    case draw
+}
+
+func getUserInput() -> Int? {
+    guard let userInput = readLine(), let userIntInput = Int(userInput)
+    else {
+        return nil
+    }
+    if userIntInput >= GameOptions.gameEnd.rawValue && userIntInput <= GameOptions.paper.rawValue {
+        return userIntInput
+    }
+    return nil
+}
+
+func checkGameResult(player playerInput: Int, computer computerInput: Int) -> GameResult {
+    if playerInput == computerInput {
+        return GameResult.draw
+    } else if (playerInput == GameOptions.scissor.rawValue && computerInput == GameOptions.lock.rawValue) ||
+              (playerInput == GameOptions.lock.rawValue && computerInput == GameOptions.paper.rawValue) ||
+              (playerInput == GameOptions.paper.rawValue && computerInput == GameOptions.scissor.rawValue)
+    {
+        return GameResult.lose
+    } else {
+        return GameResult.win
+    }
```

Comment:
- 이쪽 로직을 개선할 수 있는지 질문 주셨는데요! ㅎㅎㅎ
가위바위보 게임은 순방향으로 흐르면서도 `변칙`이 하나 존재하죠.

- 가위(1), 바위(2), 보(3) 가 있을 때
가위(1) < 바위(2)
바위(2) < 보(3)
이렇게 숫자가 크면 이길 때도 있지만
가위(1) > 보(3) 와 같은 경우엔 숫자가 작은 경우에 이기니까요 😄 
이 변칙 때문에 로직 구현이 조금 복잡해지더라구요.

- 링고와 우롱차가 만들어주신 로직도 좋은데요.
조금의 변형을 통해 `가독성`을 개선할 수 있을 것 같아요!

- 먼저, ~rawValue 사용하는 부분, 매번 쓰기엔 텍스트가 너무 길죠?
상수 선언을 통해 이 메서드 내부에서라도 짧은 단어로 대체해서 쓰면, 가독성이 크게 향상됩니다.
그리고 논리 연산자 `&&` 표현은 `컴마(,)`로 대체할 수 있어요! ㅎㅎ 

- 또한, draw 가 나오는 상황이 제일 직관적이니까 위로 올려준 점은 좋은데요!
순서 논리 상, `승리 조건`이 먼저 나오는 게 자연스러울 것 같아요.

```suggestion
    let scissor = GameOptions.scissor.rawValue
    let rock = GameOptions.rock.rawValue
    let paper = GameOptions.paper.rawValue
    
    if playerInput == computerInput {
        return .draw
    } else if playerInput == scissor, computerInput == paper ||
              playerInput == rock, computerInput == scissor ||
              playerInput == paper, computerInput == rock {
        return .win
    } else {
        return .lose
    }
```

Reply:
`승리조건`이 `패배조건`보다 먼저 오도록 변경하였고 승리를 비교하는 로직을 짧게 수식화했습니다.

```swift
func checkGameResult(player playerInput: Int, computer computerInput: Int) -> GameResult {
    let computerLoseCondition = (computerInput % 3) + 1

    if playerInput == computerLoseCondition {
        return GameResult.playerWin
    } else if playerInput == computerInput {
        return GameResult.playerDraw
    } else {
        return GameResult.playerLose
    }
}
```